Library/Homebrew/extend/os/mac/extend/ENV/super.rb

```diff
@@ -97,23 +93,24 @@ def determine_cccfg
     s.freeze
   end
 
-  def effective_sysroot
-    MacOS.sdk_path_if_needed&.to_s
-  end
-
   def set_x11_env_if_installed
     ENV.x11 = MacOS::X11.installed?
   end
 
   # @private
   def setup_build_environment(formula = nil)
-    generic_setup_build_environment(formula)
-    self["HOMEBREW_SDKROOT"] = effective_sysroot
-    self["HOMEBREW_DEVELOPER_DIR"] = if MacOS::CLT.installed? && MacOS::CLT.provides_sdk?
-      MacOS::CLT::PKG_PATH
+    if formula&.requirements&.find { |req| req.is_a? XcodeRequirement }
+      self["HOMEBREW_SDKROOT"] = MacOS.sdk_path_if_needed && MacOS::Xcode.sdk_path
```

Comment:
```suggestion
      self["HOMEBREW_SDKROOT"] = if MacOS.sdk_path_if_needed
        MacOS::Xcode.sdk_path
      end
```

does this do it? If not, the slightly gross:

```suggestion
      self["HOMEBREW_SDKROOT"] = if MacOS.sdk_path_if_needed
        MacOS::Xcode.sdk_path
      else
        nil
      end
```

Reply:
First:

```
/usr/local/Homebrew/Library/Homebrew/extend/os/mac/extend/ENV/super.rb:103:34: C: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
      self["HOMEBREW_SDKROOT"] = if MacOS.sdk_path_if_needed
```

Second:

```
/usr/local/Homebrew/Library/Homebrew/extend/os/mac/extend/ENV/super.rb:105:7: C: Redundant else-clause.
      else
      ^^^^
```

Reply:
Bah, defeated by rubocop. Ignore me!